wasmtime-environ/src/cache/config.rs

```diff
@@ -0,0 +1,621 @@
+) -> &'static Vec<String> {
+    INIT_CALLED
+        .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+        .expect("Cache system init must be called at most once");
+    assert!(
+        CONFIG.r#try().is_none(),
+        "Cache system init must be called before using the system."
+    );
+    let conf_file_str = format!("{:?}", config_file);
+    let conf = CONFIG.call_once(|| CacheConfig::from_file(enabled, config_file));
+    if conf.errors.is_empty() {
+        if conf.enabled() {
+            worker::init(init_file_per_thread_logger);
+        }
+        debug!("Cache init(\"{}\"): {:#?}", conf_file_str, conf)
+    } else {
+        error!(
+            "Cache init(\"{}\"): errors: {:#?}",
+            conf_file_str, conf.errors,
+        )
+    }
+    &conf.errors
+}
+
+/// Creates a new configuration file at specified path, or default path if None is passed.
+/// Fails if file already exists.
+pub fn create_new_config<P: AsRef<Path> + Debug>(
+    config_file: Option<P>,
+) -> Result<PathBuf, String> {
+    trace!("Creating new config file, path: {:?}", config_file);
+
+    let config_file = config_file.as_ref().map_or_else(
+        || DEFAULT_CONFIG_PATH.as_ref().map(|p| p.as_ref()),
+        |p| Ok(p.as_ref()),
+    )?;
+
+    if config_file.exists() {
+        Err(format!(
+            "Specified config file already exists! Path: {}",
+            config_file.display()
+        ))?;
+    }
+
+    let parent_dir = config_file
+        .parent()
+        .ok_or_else(|| format!("Invalid cache config path: {}", config_file.display()))?;
+
+    fs::create_dir_all(parent_dir).map_err(|err| {
+        format!(
+            "Failed to create config directory, config path: {}, error: {}",
+            config_file.display(),
+            err
+        )
+    })?;
+
+    let content = "\
+# Comment out certain settings to use default values.
+# For more settings, please refer to the documentation:
+# https://github.com/CraneStation/wasmtime/blob/master/CACHE_CONFIGURATION.md
+
+[cache]
+enabled = true
+";
+
+    fs::write(&config_file, &content).map_err(|err| {
+        format!(
+            "Failed to flush config to the disk, path: {}, msg: {}",
+            config_file.display(),
+            err
+        )
+    })?;
+
+    Ok(config_file.to_path_buf())
+}
+
+// permitted levels from: https://docs.rs/zstd/0.4.28+zstd.1.4.3/zstd/stream/write/struct.Encoder.html
+const ZSTD_COMPRESSION_LEVELS: std::ops::RangeInclusive<i32> = 0..=21;
+lazy_static! {
+    static ref PROJECT_DIRS: Option<ProjectDirs> =
+        ProjectDirs::from("", "CraneStation", "wasmtime");
+    static ref DEFAULT_CONFIG_PATH: Result<PathBuf, String> = PROJECT_DIRS
+        .as_ref()
+        .map(|proj_dirs| proj_dirs.config_dir().join("wasmtime-cache-config.toml"))
+        .ok_or("Config file not specified and failed to get the default".to_string());
+    static ref NUM_UNIT_REGEX: Regex = Regex::new(r"^\s*(\d+)\s*([^\s]*)\s*$").unwrap();
```

Comment:
Can we manually parse the num/unit string instead of pulling in regex as a dependnency?